shared/src/amm_pair_provider.rs

```diff
@@ -27,6 +30,17 @@ impl AmmPairProvider for UniswapPairProvider {
     }
 }
 
+pub struct SushiswapPairProvider {
+    pub factory: SushiswapV2Factory,
+    pub chain_id: u64,
```

Comment:
how are you using chain_id?

Reply:
It's only used to redirect the uniswap pairs to honeyswap on xdai.

Reply:
Something tells me you are starting to agree that the chain id isn't really natural to include as part of this.

Reply:
This isn't used by `SushiswapPairProvider` though... Maybe this was a copy-pasta error from the `UniswapPairProvider`?

I think, since there is an `AmmPairProvider` trait we should have:
```
pub struct UniswapPairProvider;
pub struct HoneyswapPairProvider;
pub struct SushiswapPairProvider;
```

And then select which ones to use based on the network.

Reply:
You need it for the UniswapPairProvider, but why it is added for SushiswapPairProvider if it is not used?

Reply:
> why it is added for SushiswapPairProvider if it is not used?

Right, for some reason I thought I needed to follow the same pattern. Will remove.

Reply:
> I think, since there is an AmmPairProvider trait we should have:
pub struct UniswapPairProvider;
pub struct HoneyswapPairProvider;
pub struct SushiswapPairProvider;

I think this a good suggestion for further improvement. For now we consider Honeyswap as "The Uniswap on xDAI" (but the logic of which UniswapLike liquidity to use could also be done in main).